train.py

```diff
@@ -1 +1,241 @@
+        Models saving directory.
+
+    :param logs_dir:
+        Logs saving directory.
+
+    :return:
+        Randomly initialised model prepared for training.
+        Checkpointer object containing parameters needed in model saving.
+        Tensorboard object containing parameters needed in logs saving.
+    """
+    model = Sequential()
+
+    model.add(Convolution2D(32, (3, 3), activation='relu', input_shape=(shape, shape, 1)))
+
+    model.add(Convolution2D(32, (3, 3), activation='relu'))
+    model.add(MaxPooling2D(pool_size=(2, 2)))
+    model.add(Dropout(0.25))
+
+    model.add(Flatten())
+    model.add(Dense(128, activation='relu'))
+    model.add(Dropout(0.5))
+    model.add(Dense(2, activation='softmax'))
+
+    opt = optimizers.Adam(lr=learning_rate)
+    model.compile(loss='categorical_crossentropy', optimizer=opt, metrics=['accuracy'])
+
+    checkpointer = ModelCheckpoint(filepath=models_dir,
+                                   verbose=1, save_best_only=False, save_weights_only=False, period=saving_period)
+
+    tensorboard = TensorBoard(log_dir=logs_dir.format(time()))
+
+    return model, checkpointer, tensorboard
+
+
+def train(batch_size, epochs, model, checkpointer, tensorboard, x_train, y_train, x_validate, y_validate):
+    """
+    This module trains neural network model.
+
+    :param batch_size:
+        Number of images in one training batch.
+
+    :param epochs:
+        Number of training epochs.
+
+    :param model:
+        Neural network model using in training process.
+
+    :param checkpointer:
+        Object containing parameters needed in model saving.
+
+    :param tensorboard:
+        Object containing parameters needed in logs saving.
+
+    :param x_train:
+        Input training data.
+
+    :param y_train:
+        Output training data.
+
+    :param x_validate:
+        Input validation data.
+
+    :param y_validate:
+        Output validation data.
+
+    :return:
+        Trained neural network model.
+    """
+    model.fit(x_train, y_train,
+              batch_size=batch_size, epochs=epochs, verbose=1, shuffle=False,
+              validation_data=(x_validate, y_validate), callbacks=[checkpointer, tensorboard])
+
+    return model
+
+
+def test(epochs, saving_period, x_test, y_test, load_dir, model):
+    """
+    This module tests model after each training epoch on testing dataset and shows score.
+
+    :param epochs:
+        Number of training epochs.
+
+    :param saving_period:
+        How many epochs of training must pass before saving model.
+
+    :param x_test:
+        Input testing data.
+
+    :param y_test:
+        Output testing data.
+
+    :param load_dir:
+        Path for loading model.
+
+    :param model
+        Tested  neural network model.
+    """
+    for i in range(int(epochs/saving_period)):
+        # Downloading weights
+        try:
+            model = load_model(load_dir.format(i+1))
+        except:
+            print("Unable to load model")
```

Comment:
This looks very bad. Firstly you hide an actual error and replace it with mysterious message "Unable to load model". Secondly you proceed with the model even though you caught an error.